From 92508533134fa52eed879cb3204836fd4e2e6027 Mon Sep 17 00:00:00 2001
From: Ihor Ivanov <toulmee@gmail.com>
Date: Sun, 26 Sep 2021 20:12:54 +0300
Subject: [PATCH 01/39] camera2: Add methods for backward compatibility

libmcf-jni.so library (which is proprietary blob required
for MotCamera2) looks for methods with these exact signatures,
but signatures of these methods was changed in Android 11.

This change returns these methods that makes MotCamera2
fully-functional on Android 11.

Change-Id: Ib9a701f6d8ebb783c52666e34e7539faba5cb99f
Signed-off-by: Ihor Ivanov <toulmee@gmail.com>
---
 .../hardware/camera2/impl/CameraMetadataNative.java  | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/core/java/android/hardware/camera2/impl/CameraMetadataNative.java b/core/java/android/hardware/camera2/impl/CameraMetadataNative.java
index 5da8d5436274..eb1268657a6f 100644
--- a/core/java/android/hardware/camera2/impl/CameraMetadataNative.java
+++ b/core/java/android/hardware/camera2/impl/CameraMetadataNative.java
@@ -2352,6 +2352,18 @@ public class CameraMetadataNative implements Parcelable {
     private static native int nativeGetTypeFromTag(int tag, long vendorId)
             throws IllegalArgumentException;
 
+    private synchronized byte[] nativeReadValues(int tag) {
+        return nativeReadValues(tag, mMetadataPtr);
+    }
+
+    private synchronized int nativeGetTypeFromTagLocal(int tag) {
+        return nativeGetTypeFromTagLocal(mMetadataPtr, tag);
+    }
+
+    private synchronized int nativeGetTagFromKeyLocal(String keyname) {
+        return nativeGetTagFromKeyLocal(mMetadataPtr, keyname);
+    }
+
     /**
      * <p>Perform a 0-copy swap of the internal metadata with another object.</p>
      *
-- 
2.43.0

